interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -163,12 +165,36 @@ class NodeCompiler(
           )
       }
 
-      val parameterExtractionValidation =
-        NonEmptyList.fromList(parameterDefinitions.written).map(invalid).getOrElse(valid(List.empty))
+      val parameterNameValidation = parameterDefinitions.value
```

Comment:
i'd move this logic to a helper function, this function is complicated enough as is

Reply:
Good point, done :+1: